### PR TITLE
pickupEnergy with FIND_TOMBSTONES

### PR DESCRIPTION
**Pull request by mschultheiss83**

### CODE REVIEW

**Comment by mschultheiss83** [comment]
@TooAngel please help on eslint:check FIND_TOMBSTONES

**Comment by TooAngel** [comment]
Sure https://github.com/PostCrafter/eslint-plugin-screeps/pull/4

File: src/prototype_creep_resources.js [hunk 1]
```diff
@@ -280,6 +280,26 @@ Creep.prototype.pickupEnergy = function() {
     }
   }
 
+  // FIND_TOMBSTONES and get them empty first
```
Comment (TooAngel) [inline src/prototype_creep_resources.js]: We should create a new function for it `Creep.prototype.withdrawTombstone` and call it from the `getEnergy()` function.

File: src/prototype_creep_resources.js [hunk 2]
```diff
@@ -280,6 +280,26 @@ Creep.prototype.pickupEnergy = function() {
...
+  const tombstones = this.pos.findInRange(FIND_TOMBSTONES, 1, {
+    filter: (t) => {
+      return t.creep.my;
```
Comment (TooAngel) [inline src/prototype_creep_resources.js]: I think you can also withdraw from others, so you don't need to filter.

File: src/prototype_creep_resources.js [hunk 3]
```diff
@@ -280,6 +280,26 @@ Creep.prototype.pickupEnergy = function() {
...
+    },
+  });
+  if (tombstones.length > 0) {
+    const creep = this;
+    _.map(tombstones, (t) => {
+      if (t.store[RESOURCE_ENERGY] > 0) {
```
Comment (TooAngel) [inline src/prototype_creep_resources.js]: Why only energy?

File: src/prototype_creep_resources.js (hunk 2 above)
Comment (mschultheiss83) [reply src/prototype_creep_resources.js]: good point

File: src/prototype_creep_resources.js (hunk 1 above)
Comment (mschultheiss83) [reply src/prototype_creep_resources.js]: i will do

File: src/prototype_creep_resources.js (hunk 3 above)
Comment (mschultheiss83) [reply src/prototype_creep_resources.js]: it was in the getEnergy function... but a onlyEnergy flag i would like, or why would a sourcer on his way pick up resources?

Comment (TooAngel) [reply src/prototype_creep_resources.js]: Yeah, good point

**Comment by mschultheiss83** [comment]
can we merge this?

**Comment by mschultheiss83** [comment]
so with the ´withdrawResourcesFromTargets´ the end of `role.storagefiller.action` will be better to read

**Comment by democratic-collaboration** [comment]
A new review, yeah.

            Votes: 52/617
            Coefficient: 0.08427876823338736
            Merging in 15 days 23.630277777777778 hours
            Age 2 days 7.915555555555556 hours

**Pull request merged**